File: exercism/go/archive/allergies/allergies.go
// Package allergies returns the list of ailments from a given score
package allergies

import (
	"fmt"
	"strconv"
)

// testVersion is the suite we're up against
const testVersion = 1

var items = [8]string{
	"eggs",
	"peanuts",
	"shellfish",
	"strawberries",
	"tomatoes",
	"chocolate",
	"pollen",
	"cats",
}

// Allergies returns the list of ailments given an allergy "score"
func Allergies(input uint) (ailments []string) {
	score := fmt.Sprintf("%08s", strconv.FormatUint(uint64(input%256), 2))

	for x := len(score) - 1; x >= 0; x-- {
		if score[x] == '1' {
			ailments = append(ailments, items[len(score)-x-1])
		}
	}

	return ailments
}

// AllergicTo returns whether the given allergy "score" matches the ailment
func AllergicTo(input uint, ailment string) bool {
	for _, each := range Allergies(input) {
		if each == ailment {
			return true
		}
	}

	return false
}

File: exercism/python/archive/allergies/allergies.py
""" Module to provide the list of alleriges to which the user is affected """
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

class Allergies(object):
    """ base class for our allergy set """
    def __init__(self, score):
        """ init function whereby we assign the user's allegies """
        self.bitwise_score = score % 256
        self.conditions = {
            0: "eggs",
            1: "peanuts",
            2: "shellfish",
            3: "strawberries",
            4: "tomatoes",
            5: "chocolate",
            6: "pollen",
            7: "cats",
        }

        self.ailments = []

        for each in range(8):
            if self.bitwise_score & (1 << each):
                self.ailments.append(self.conditions[each])

    def is_allergic_to(self, src=""):
        """ check if we're allergic to the given cause """
        if not src:
            return False

        for each in range(8):
            if self.conditions[each] != src:
                continue

            if self.bitwise_score & (1 << each):
                return True

        return False

    @property
    def lst(self):
        """ based on the score provided, return the list of ailments """
        return self.ailments
